### PR TITLE
perf(object-hash): reuse default options when is not passed

### PR DESCRIPTION
**Pull request by H4ad**
Just a tiny performance improvement to avoid allocate more memory and doing some spreading that is not necessary.

### CODE REVIEW

File: src/object-hash.ts [hunk 1]
```diff
@@ -61,8 +64,9 @@ const defaults: HashOptions = {
  * @return {string} hash value
  * @api public
  */
-export function objectHash(object: any, options: HashOptions = {}): string {
-  options = { ...defaults, ...options };
+export function objectHash(object: any, options?: HashOptions): string {
+  if (!options) options = defaults;
```
Comment (pi0) [inline src/object-hash.ts:68]: It would be nice if make sure defaults is frozen with `Object.freeze` to not accidentally modify it

Comment (H4ad) [reply src/object-hash.ts:68]: I thought about doing this, but I didn't see any code exposing `defaults`, so I didn't think it was necessary.

Comment (pi0) [reply src/object-hash.ts:68]: Internally mainly i want to avoid any (future) regressions to unwillingly modify defaults

Comment (H4ad) [reply src/object-hash.ts]: Okay, pushed!

**Pull request merged**